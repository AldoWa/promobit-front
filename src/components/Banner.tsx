import React from 'react';

import Filter from './Filter';

import { Box, Container, VStack, Text } from '@chakra-ui/react';

const Banner: React.FC = () => {
  return (
    <Box height={449} bg='purple.900'>
      <Container maxW={1033} height='100%' alignItems='center'>
        <VStack
          spacing={10}
          height='100%'
          justifyContent='center'
        >
          <Text fontSize='5xl' color='white' fontWeight='bold' textAlign={'center'} lineHeight='56px'>
            Milhões de filmes, séries e pessoas para descobrir. Explore já.
          </Text>
          <Filter />
        </VStack>
      </Container>
    </Box>
  );
};

export default Banner;
