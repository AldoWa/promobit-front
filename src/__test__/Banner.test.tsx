import { render, screen } from '@testing-library/react';
import Banner from '../components/Banner';

jest.mock('../services/api.ts', () => {
	return {
    AUTHORIZATION: 'XXXX'
  }
})

jest.mock('../components/Filter')

describe('Banner', () => {
  test("Should render the banner", () => {
    render(<Banner />)
    const element = screen.getByTestId('banner-box')
    expect(element).toBeInTheDocument()
    expect(element).toBeVisible()
  })

  test("Should render banner text", () => {
    render(<Banner />)
    const element = screen.getByTestId('banner-text')
    expect(element).toBeInTheDocument()
    expect(element).toBeVisible()
    expect(element).toHaveTextContent('Milhões de filmes, séries e pessoas para descobrir. Explore já.')
  })

  test("Should match snapshot", () => {
    const { container } = render(<Banner />)
    expect(container).toMatchSnapshot();
  })
})