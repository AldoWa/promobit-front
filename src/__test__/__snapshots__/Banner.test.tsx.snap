// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Should match snapshot 1`] = `
<div>
  <div
    class="css-shrtae"
    data-testid="banner-box"
  >
    <div
      class="chakra-container css-1hs5bnt"
    >
      <div
        class="chakra-stack css-o9yjr2"
      >
        <p
          class="chakra-text css-p7o7s7"
          data-testid="banner-text"
        >
          Milhões de filmes, séries e pessoas para descobrir. Explore já.
        </p>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
